test/unit/pipeline/DispatchIssueUnitTest.cc

```diff
@@ -0,0 +1,543 @@
+  EXPECT_EQ(diUnit.getFrontendStalls(), 1);
+  EXPECT_EQ(diUnit.getBackendStalls(), 0);
+  EXPECT_EQ(diUnit.getPortBusyStalls(), 0);
+}
+
+// Single instruction has no exception, 2 source operands (both ready), 1
+// destination operand
+TEST_F(PipelineDispatchIssueUnitTest, singleInstr) {
+  // Set-up source & destination registers and ports for this instruction
+  std::array<Register, 2> srcRegs = {r1, r2};
+  std::array<Register, 1> destRegs = {r0};
+  const std::vector<uint16_t> suppPorts = {EAGA};
+
+  // All expected calls to instruction during tick()
+  EXPECT_CALL(*uop, getSupportedPorts()).WillOnce(ReturnRef(suppPorts));
+  uop->setExceptionEncountered(false);
+  EXPECT_CALL(*uop, getSourceRegisters())
+      .WillOnce(Return(span<Register>(srcRegs)));
+  EXPECT_CALL(*uop, isOperandReady(0)).WillOnce(Return(false));
+  EXPECT_CALL(*uop, supplyOperand(0, RegisterValue(0, 8)));
+  EXPECT_CALL(*uop, isOperandReady(1)).WillOnce(Return(false));
+  EXPECT_CALL(*uop, supplyOperand(1, RegisterValue(0, 8)));
+  EXPECT_CALL(*uop, getDestinationRegisters())
+      .WillOnce(Return(span<Register>(destRegs)));
+
+  // Expected call to port allocator during tick()
+  EXPECT_CALL(portAlloc, allocate(suppPorts)).WillOnce(Return(EAGA));
+
+  // Ensure empty reservation stations pre tick()
+  std::vector<uint64_t> rsSizes;
+  diUnit.getRSSizes(rsSizes);
+  EXPECT_EQ(rsSizes, refRsSizes);
+
+  input.getHeadSlots()[0] = uopPtr;
+  diUnit.tick();
+  // Ensure post tick that EAGA's reservation station size has decreased by 1
+  rsSizes.clear();
+  diUnit.getRSSizes(rsSizes);
+  EXPECT_EQ(rsSizes.size(), refRsSizes.size());
+  EXPECT_EQ(rsSizes[RS_EAGA], refRsSizes[RS_EAGA] - 1);
+  // Ensure no stalls recorded in tick()
+  EXPECT_EQ(diUnit.getFrontendStalls(), 0);
+  EXPECT_EQ(diUnit.getBackendStalls(), 0);
+  EXPECT_EQ(diUnit.getPortBusyStalls(), 0);
+  EXPECT_EQ(diUnit.getRSStalls(), 0);
+  // Ensure empty output buffers post tick()
+  for (size_t i = 0; i < output.size(); i++) {
+    EXPECT_EQ(output[i].getTailSlots()[0], nullptr);
+  }
+
+  // Detail expected call to port allocator during tick()
+  EXPECT_CALL(portAlloc, issued(EAGA));
+
+  diUnit.issue();
+  // Ensure all reservation stations empty again post issue()
+  rsSizes.clear();
+  diUnit.getRSSizes(rsSizes);
+  EXPECT_EQ(rsSizes, refRsSizes);
+  // Ensure no stalls recorded during issue()
+  EXPECT_EQ(diUnit.getFrontendStalls(), 0);
+  EXPECT_EQ(diUnit.getBackendStalls(), 0);
+  EXPECT_EQ(diUnit.getPortBusyStalls(), 0);
+  EXPECT_EQ(diUnit.getRSStalls(), 0);
+  // Ensure all output buffers are empty, except the one associated with EAGA
+  // port which contains the uop
+  for (size_t i = 0; i < output.size(); i++) {
+    if (i != EAGA)
+      EXPECT_EQ(output[i].getTailSlots()[0], nullptr);
+    else
+      EXPECT_EQ(output[i].getTailSlots()[0].get(), uop);
+  }
+}
+
+// Single instruction with exception
+TEST_F(PipelineDispatchIssueUnitTest, singleInstr_excpetion) {
```

Comment:
singleInstr_exception not singleInstr_excpetion

Reply:
Fixed